app.ts

```diff
@@ -0,0 +1,168 @@
+/** @format */
+
+class DataBase implements DataAggregator {
+    private static instance: DataBase;
+    private users: any[] = [];
+
+    public static getInstance(): DataBase {
+        if (!DataBase.instance) {
+            DataBase.instance = new DataBase();
+        }
+
+        return DataBase.instance;
+    }
+
+    public addUser(user: object): void {
+        this.users.push(user);
+    }
+
+    public getUsers(): object {
+        return this.users;
+    }
+
+    public removeUser(value: string, key: string): object {
+        this.users.filter(item => item[key] !== value)
+        return this.users;
+    }
+
+    public createIterator(): DataBaseIterator {
+        return new DataBaseIterator(this)
+    }
+}
+
+interface DataAggregator {
+    addUser(user: object): void;
+    getUsers(): object;
+    removeUser(value: string, key: string): object;
+    createIterator(): DataBaseIterator;
+}
+
+class DataBaseIterator {
+    private position: number = 0;
+
+    constructor(private users: DataAggregator) { }
+
+    public current(index: number): object {
+        this.position = index;
+        return this.users.getUsers()[index];
+    }
+
+    public next(): void {
+        this.position++;
+    }
+
+    public prev(): void {
+        this.position--;
+    }
+}
+
+class User {
+    firstName: string;
+    secondName: string;
+    age: number;
+    gender: string;
+
+    constructor(
+        userFirstName: string,
+        userSecondName: string,
+        userAge: number,
+        userGender: string
+    ) {
+        this.firstName = userFirstName;
+        this.secondName = userSecondName;
+        this.age = userAge;
+        this.gender = userGender;
+    }
+
+    public addUser() {
+        DataBase.getInstance().addUser(this);
```

Comment:
We are kinda violating Single Responsibility Principle here. What if we want add the user into another data base. What if we want to add user into some TextFile. Please consider removing this method entirely, and use your DataBase singleton directly for this, or create some DataBaseProxy